src/compiler/checker.ts

```diff
@@ -34018,7 +34018,7 @@ namespace ts {
             }
             switch (kind) {
                 case SyntaxKind.Identifier:
-                    return checkIdentifier(node as Identifier, checkMode);
+                    return isThisInTypeQuery(node) ? checkThisExpression(node) : checkIdentifier(node as Identifier, checkMode);

```

Comment:
fairly minor style request: move this check into the top of `checkIdentifier`. It's only called from here, and it keeps checkExpressionWorker as simple as possible.

(Looking at the rest of checkExpressionWorker, the only exception I see is BigIntLiteral, which just has the inline source of what *should* be checkBigIntLiteral.)